subsys/fs/fs.c

```diff
@@ -75,7 +76,7 @@ static int fs_get_mnt_point(struct fs_mount_t **mnt_pntp,
 }
 
 /* File operations */
-int fs_open(struct fs_file_t *zfp, const char *file_name)
+int fs_open(struct fs_file_t *zfp, const char *file_name, unsigned int flags)
```

Comment:
The third parameter should be of type:

    typedef uint8_t fs_mode_t;

not a bare unsigned int.

Reply:
> The third parameter should be of type:
typedef uint8_t fs_mode_t;
not a bare unsigned int.

Perhaps, but then it should be typedef to the very unsigned int, not an adhoc type like uint8_t. When passing params to funcs, native word sized params should be preferred, unless there's a specific need to do otherwise.

Reply:
In POSIX the value passed through this argument is documented to be of type `mode_t`.

Since the FS API is not a POSIX API [coding guidelines Dir 4.6](https://docs.zephyrproject.org/latest/guides/coding_guidelines/index.html) suggest that it should be a standard sized type.

I don't particularly care what it's typedef'd as, but it needs to be a typedef'd type.

Reply:
OK, do you want it to be `uint8_t`? If ABI passes it through registers it will be of register size anyway (which is usually int for CPU) or passed via stack that is usually aligned to size of CPU register. Why to restrict it to `uint8_t`

Reply:
WIll make it `uint8_t`, the ",member of structure" argument by @pabigot is ok.

Reply:
Done.